cpp/include/raft/core/operators.hpp

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <cmath>
+
+#include <raft/core/detail/macros.hpp>
+
+namespace raft {
+
+/**
+ * @defgroup Functors Commonly used functors.
+ * The optional index argument is mostly to be used for MainLambda in reduction kernels
+ * @{
+ */
+
+struct identity_op {
+  template <typename Type, typename IdxType = int>
```

Comment:
Would it make sense to replace the `IdxType` here and in the other operators, which do not use the second argument, with something like the this?:
```
template <typename Type, typename UnusedArgs...>
constexpr RAFT_INLINE_FUNCTION auto operator()(Type in, UnusedArgs...) const { ... }
```
I think, the latter would be more flexible and explicit in the fact that it does not use and does not need any sorts of indices.

Reply:
Good point! I've been quite conservative here but I like your idea better.